src/routes/products/idSearcher.js

```diff
@@ -0,0 +1,35 @@
+const products = require("../../db/products/all-products.json");
+const url = require("url");
+
+const idSearcher = (request, response) => {
+  const urlObj = url.parse(request.url);
+  console.log("urlObj :", urlObj);
+  const idFromPath = getIdFromPath(urlObj.path);
+  const idProd = products.filter(el => el.id === Number(idFromPath));
+  let foundedProducts;
+
+  if (idProd.length === 0 || !idProd) {
```

Comment:
in your case, `idProd` would be always array. So condition `!idProd` would be never met